metadata/src/main/java/org/apache/kafka/image/TopicsDelta.java

```diff
@@ -54,6 +55,7 @@ public final class TopicsDelta {
      */
     private final Set<Uuid> deletedTopicIds = new HashSet<>();
 
+    private final Map<String, Uuid> createdTopics = new HashMap<>();
```

Comment:
I don't see any place where we are using `createdTopicIds` as a set. We just use it as a collection (to check if it's empty?) I suggest changing the accessor function to return a `Collection<Uuid>` instead of a `Set<Uuid>`, and getting rid of `final Set<Uuid> createdTopicIds` to avoid bloating memory consumption.